packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/RealmImpl.kt

```diff
@@ -131,13 +130,14 @@ public class RealmImpl private constructor(
                 }
                 val (frozenReference, fileCreated) = configuration.openRealm(this@RealmImpl)
                 realmFileCreated = assetFileCopied || fileCreated
-                versionTracker.trackAndCloseExpiredReferences(frozenReference)
-                _realmReference.value = frozenReference
+                versionTracker.trackReference(frozenReference)
+                initialRealmReference.value = frozenReference
                 configuration.initializeRealmData(this@RealmImpl, realmFileCreated)
             }
 
             realmScope.launch {
                 notifier.realmChanged().collect {
+                    removeInitialRealmReference()
```

Comment:
This is executing on the `notifier` thread so you need to synchronize it to be sure that it is up to date.

Reply:
Would it somehow be possible to pass the initial frozen version into the notifier's version tracker instead? Then we could completely eliminate the version tracker here 🤔

Reply:
That was my initial design, but then it kind of convolute testing on the notifier version tracker.

Because the notifier loads later, you might find one or two active versions, user-facing and notifier pending to be released.